flake.nix

```diff
@@ -233,31 +241,95 @@
                 mv $out/usr/etc $out/etc
               '';
             };
+
+            version =
+              if name == "wasmcloud"
+              then (readTOML ./Cargo.toml).package.version
```

Comment:
Help me understand why we read from the Cargo.toml instead of using the git ref?

Reply:
The version in `Cargo.toml` must match the version of the binary and it's the simplest approach

Reply:
For reference, here's the spec on these https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys, which is very loosely defined (which is generally true for all of these labels and they really should not be used as an API of any sorts)

>    The version MAY match a label or tag in the source code repository
>    version MAY be Semantic versioning-compatible